code/modules/power/supermatter/supermatter.dm

```diff
@@ -684,6 +684,21 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 
 	Consume(AM)
 
+/obj/machinery/power/supermatter_crystal/intercept_zImpact(atom/movable/AM, levels)
+	if(isliving(AM))
+		AM.visible_message("<span class='danger'>\The [AM] slams into \the [src] inducing a resonance... [AM.p_their()] body starts to glow and burst into flames before flashing into dust!</span>",\
+		"<span class='userdanger'>You slam into \the [src] as your ears are filled with unearthly ringing. Your last thought is \"Oh, fuck.\"</span>",\
+		"<span class='hear'>You hear an unearthly noise as a wave of heat washes over you.</span>")
+	else if(isobj(AM) && !iseffect(AM))
+		AM.visible_message("<span class='danger'>\The [AM] smacks into \the [src] and rapidly flashes to ash.</span>", null,\
+		"<span class='hear'>You hear a loud crack as you are washed with a wave of heat.</span>")
+	else
+		return FALSE
+
+	playsound(get_turf(src), 'sound/effects/supermatter.ogg', 50, TRUE)
+
+	Consume(AM)
+
 /obj/machinery/power/supermatter_crystal/proc/Consume(atom/movable/AM)
```

Comment:
Isn't this the same as otherwise touching it?

Reply:
yes, I guess I could just make it call the proc for touching it.